src/imagej/convert.py

```diff
@@ -458,6 +458,75 @@ def supports_imglabeling_to_labeling(obj):
     return isinstance(obj, jc.ImgLabeling)
 
 
+#######################
+# IndexImg converters #
+#######################
+
+
+def index_img_to_roi_tree(
+    ij: "jc.ImageJ", index_img: Union[np.ndarray, "jc.RandomAccessibleInterval"]
+) -> "jc.DefaultROITree":
+    """
+    Convert an ImgLib2 or NumPy index image to a ROITree.
+
+    :param ij: The ImageJ2 gateway (see imagej.init)
+    :param index_img: An ImgLib2 or NumPy index image
+    :return: A ROITree with ImgLib2 ROIs
+    """
+    # convert ImgLib2 IndexImgs to NumPy
+    if sj.isjava(index_img):
+        index_img = ij.py.from_java(index_img)
+
+    # get contours and populate a ROITree
+    rois = _get_contours(ij, index_img)
+    roi_tree = jc.DefaultROITree()
+    roi_tree.addROIs(jc.ArrayList(rois))
+
+    return roi_tree
+
+
+def index_img_to_roi_manager(
+    ij: "jc.ImageJ", index_img: Union[np.ndarray, "jc.RandomAccessibleInterval"]
+):
+    """
+    Convert an ImgLib2 or NumPy into ImageJ ROIs and populate
+    the RoiManager.
+
+    :param ij: The ImageJ2 gateway (see imagej.init)
+    :param index_img: An ImgLib2 or NumPy index image
+    """
+    # convert ImgLib2 IndexImgs to NumPy
+    if sj.isjava(index_img):
+        index_img = ij.py.from_java(index_img)
+
+    # get contours and populate the ROIManager
+    rois = _get_contours(ij, index_img)
+    try:
+        rm = ij.RoiManager.getRoiManager()
+        for r in rois:
+            rm.addRoi(ij.convert().convert(r, jc.PolygonRoi))
+    except JException:
+        print("The RoiManager is unavailable in headless mode.")
+
+
+def _get_contours(ij: "jc.ImageJ", index_img: np.ndarray) -> List:
+    """
+    Compute contours from an index image.
+
+    :param ij: The ImageJ2 gateway (see imagej.init)
+    :param index_img: An ImgLib2 or NumPy index image
+    """
+    contours = []
+    labels = np.unique(index_img)
+    for i in range(len(labels)):
+        if labels[i] == 0:
+            continue
+        mask = index_img == labels[i]
```

Comment:
This feels really inefficient to me...have you noticed any performance issues here? Of course, let's not prematurely optimize.

One thing that could help is [`ImgLabeling`](https://github.com/imglib/imglib2-roi/blob/57e4036c9a420dd13a6f7aff87aa793b76658e54/src/main/java/net/imglib2/roi/labeling/ImgLabeling.java#L90), which would create an `ImgLabeling` from an index image - then maybe you could use `LabelRegions` to get the regions from the `ImgLabeling`? Not sure if the regions are rich enough for your purpose, though...

Reply:
Yeah I kind of felt the same way when I wrote this. It hasn't been an issue with the datasets I'm working with but I would really question how well this would scale for really large datasets, or index images that have a lot of objects. I thought about using `ImgLabeling` but I wasn't sure if I wanted to tango with `ImgLib2`.

I'm going to see if I can write this method using `ImgLabeling` and `LabelRegions` and then do a performance comparison.

Reply:
I re-wrote this function using `ImgLabeling` and `LabelRegions` in this commit: edbda0af121236f7e23fa2c59776b1baab7d9451. I did some rough testing and its 5.73 times _faster_ than the NumPy implementation. :+1:  Good stuff!